ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -364,15 +354,16 @@ public StreamPullQueryMetadata createStreamPullQuery(
             .putAll(statementOrig.getSessionConfig().getOverrides())
             .put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
             .put(StreamsConfig.NUM_STREAM_THREADS_CONFIG, 1)
-            .put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, 100)
+            // There's no point in EOS, since this query only produces side effects.
+            .put(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, StreamsConfig.AT_LEAST_ONCE)
             .build()
     );
+    final ImmutableMap<TopicPartition, Long> endOffsets =
+        getQueryInputEndOffsets(analysis, serviceContext.getAdminClient());
+
     final TransientQueryMetadata transientQueryMetadata = EngineExecutor
         .create(primaryContext, serviceContext, statement.getSessionConfig())
-        .executeTransientQuery(statement, excludeTombstones);
-
-    final ImmutableMap<TopicPartition, Long> endOffsets =
-        getQueryInputEndOffsets(analysis, statement, serviceContext.getAdminClient());
+        .executeTransientQuery(statement, excludeTombstones, Optional.of(endOffsets));
```

Comment:
This makes me wonder: are we categorizing `push` and `pull` queries both as part of `transient` queries now? In the past `push == transient`, and `pull` queries are another category. But now since we are executing `stream pull queries` as a transient query. Or are we consider the `stream pull queries` still as part of the `push == transient` queries but just expose them as a `pull` query to users?